elastic/e2e-testing
# chore: use preconfigured policy for fleet-server's default policy

## What does this PR do?
This PR adds preconfigured policies for Fleet Server in all kibana.config.yml files used in the tests (the default and the one for the pre-configured policies scenario)

Because we are setting the fleet-server policy from the config, we are now hardcoding the policy name and ID in the bootstrap process of fleet-server (that happens right after the stack starts), not getting it from Kibana API for a default policy.

We are also adding a deprecation comment in the GetDefaultPolicy method, as it's not used anymore. We could simply remove it, but I prefer doing it in a separate PR.



## Why is it important?
Fleet team is removing default policies



## Checklist




## Author's Checklist



## How to test this PR locally
Because the current stack version is not updated, we need to set a more recent one (at least 4 days ago, where a bug was fixed on Kibana side https://github.com/elastic/kibana/pull/124363):

```shell
$ TAGS="blah" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE DEVELOPER_MODE=true ELASTIC_APM_ACTIVE=false STACK_VERSION=8.1.0-98e72485-SNAPSHOT BEAT_VERSION=8.1.0-98e72485-SNAPSHOT PROVIDER=docker make -C e2e/_suites/fleet functional-test^C
```

The `blah` tag is a non-existing one to avoid launching any scenario, but instead simply bootstrapping the stack, which happens in the before suite hook. The result would be the 3 containers for the stack (es, kbn and fleet-server) without log errors, and after checking the Kibana UI, verify that fleet-server is enrolled.



## Related issues


- Relates https://github.com/elastic/kibana/pull/124363
- Closes https://github.com/elastic/e2e-testing/issues/2039
